### PR TITLE
Add rustler precompiled

### PR DESCRIPTION
**Pull request by SteffenDE**
This adds a workflow to precompile the nifs using rustler_precompiled.

Publishing to hex is removed from the pipeline, as the checksum generation step is needed, see: https://hexdocs.pm/rustler_precompiled/precompilation_guide.html#content and especially:
https://hexdocs.pm/rustler_precompiled/precompilation_guide.html#the-release-flow

When running locally and wanting to force the build, set `LINGUA_BUILD=1`, e.g. `LINGUA_BUILD=1 mix test` or `LINGUA_BUILD=1 mix compile`.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -30,7 +35,8 @@ defmodule Lingua.MixProject do
         "README.md",
         "LICENSE",
         "native/lingua_nif/Cargo.*",
-        "native/lingua_nif/src"
+        "native/lingua_nif/src",
+        "checksum-*.exs"
```
Comment (SteffenDE) [inline mix.exs:39]: This should be generated after the precompiled artifacts are available on GitHub using:
`mix rustler_precompiled.download Lingua.Nif --all --print`

Afterwards the package can be published to hex.

**Comment by minidfx** [comment]
Any news about this PR?

**Comment by minidfx** [comment]
The nif.ex doens't found the precompiled binaries.

```elixir
== Compilation error in file lib/lingua/nif.ex ==
** (RuntimeError) Error while downloading precompiled NIF: couldn't fetch NIF from https://github.com/joshrotenberg/lingua_ex/releases/download/v0.2.0/lingua_nif-v0.2.0-nif-2.16-x86_64-pc-windows-msvc.dll.tar.gz: {:ok, {{~c"HTTP/1.1", 404, ~c"Not Found"}, [{~c"cache-control", ~c"no-cache"}, {~c"date", ~c"Thu, 21 Mar 2024 08:46:17 GMT"}, {~c"server", ~c"GitHub.com"}, {~c"vary", ~c"X-PJAX, X-PJAX-Container, Turbo-Visit, Turbo-Frame, Accept-Encoding, Accept, X-Requested-With"}, {~c"content-length", ~c"9"}, {~c"content-type", ~c"text/plain; charset=utf-8"}, {~c"strict-transport-security", ~c"max-age=31536000; includeSubdomains; preload"}, {~c"x-frame-options", ~c"deny"}, {~c"x-content-type-options", ~c"nosniff"}, {~c"x-xss-protection", ~c"0"}, {~c"referrer-policy", ~c"no-referrer-when-downgrade"}, {~c"content-security-policy", ~c"default-src 'none'; base-uri 'self'; connect-src 'self'; form-action 'self'; img-src 'self' data:; script-src 'self'; style-src 'unsafe-inline'"}, {~c"x-github-request-id", ~c"D7B9:10B68D:309F1CF:317070F:65FBF3D9"}], "Not Found"}}
```

**Comment by SteffenDE** [comment]
@minidfx if you want to use the precompiled NIFs from the PR, you need to use

```
{:lingua, github: "SteffenDE/lingua_ex", branch: "development"},
```

**Pull request merged**